src/Internal/LoggerContext.php

```diff
@@ -124,23 +124,17 @@ public function filter_log_entry( $entry, $context ): string {
 			$this->context_updated = false;
 		}
 
-		return implode(
-			"\n",
-			array_map(
-				function ( $entry ) use ( $format_string ) {
-					return implode(
-						"\n",
-						array_map(
-							function ( $line ) use ( $format_string ) {
-								return sprintf( $format_string, $line );
-							},
-							explode( "\n", $entry )
-						)
-					);
-				},
-				$entries
-			)
-		);
+		$formatted_lines = [];
+		$log_entry       = array_shift( $entries );
+		while ( null !== $log_entry ) {
+			foreach ( explode( "\n", $log_entry ) as $line ) {
+				$formatted_lines[] = $line_prefix . $line;
+			}
+			unset( $log_entry );
+			$log_entry = array_shift( $entries );
+		}
+
+		return implode( "\n", $formatted_lines );
```

Comment:
Personally, I prefer to avoid `while` loops if an iterator is available. In this case, a simple `foreach` loop should do the trick. Even if there is context here, the `$entries` array should be quite short, so this seems overoptimized, especially considering that `array_shift` will modify the array every time.

```suggestion
		$formatted_lines = [];
		foreach ( $entries as $log_entry ) {
			foreach ( explode( "\n", $log_entry ) as $line ) {
				$formatted_lines[] = $line_prefix . $line;
			}
		}
		return implode( "\n", $formatted_lines );
```

For me, the decreased number of anonymous functions created, combined with much fewer `sprintf()` calls, should be more than sufficient.

Reply:
`foreach` was my first choice. But I had to add an `unset` because I caught a memory overflow on my local setup when testing a complex transactions page. This eliminated the option of using `foreach` 😞

Reply:
Huh, could it be that you were still using `sprintf()` for the whole prefix? I'm really surprised that PHP would not collect the `foreach()` garbage. Anyhow, let's unblock others, we could experiment with this later.